Package/CreatePackage/Package.cs

```diff
@@ -673,10 +680,57 @@ private static void UpdateVersionInfo(string tempDir, List<PackageFile> files, S
             log.Info(timer,"Updated assembly version info using Mono method.");
         }
 
+        private static bool VerifyArchiveIntegrity(FileStream fs)
+        {
+            log.Debug($"Verifying package integrity.");
+            // Ensure the stream is fully written to the disk
+            fs.Flush();
+
+            {   // Verify that we can read the package definition
+                try
+                {
+                    PackageDef.FromPackage(fs.Name);
+                }
+                catch (Exception ex)
+                {
+                    log.Error($"Error verifying package integrity.");
+                    log.Debug(ex);
+                    return false;
+                }
+            }
+
+            // Rewind the stream
+            fs.Seek(0, SeekOrigin.Begin);
+            {   // Verify that we can extract the archive without errors
+                using var ms = new MemoryStream();
```

Comment:
A memoryStream cannot contain elements larger than 1G. Trying that would cause an exception. So maybe we should use Stream.Null instead?

Reply:
That's a great idea.